makefile.shade

```diff
@@ -780,4 +841,163 @@ functions
+                var srcDir = Path.Combine(repo, "src");
+
+                if (Directory.Exists(srcDir))
+                {
+                    foreach (var directory in Directory.EnumerateDirectories(srcDir))
+                    {
+                        info.RepositoryNames.Add(Path.GetFileName(directory));
+                        var projectJson = Path.Combine(directory, "project.json");
+                        if (File.Exists(projectJson))
+                        {
+                            GetDependencies(projectJson, info.DependencyNames);
+                        }
+                    }
+                }
+
+                var otherDirs = new[] { "test", "samples" };
+                for (var i = 0; i < otherDirs.Length; i++)
+                {
+                    var otherDir = Path.Combine(repo, otherDirs[i]);
+                    if (Directory.Exists(otherDir))
+                    {
+                        foreach (var directory in Directory.EnumerateDirectories(otherDir))
+                        {
+                            var projectJson = Path.Combine(directory, "project.json");
+                            if (File.Exists(projectJson))
+                            {
+                                GetDependencies(projectJson, info.DependencyNames);
+                            }
+                        }
+                    }
+                }
+
+                info.DependencyNames.ExceptWith(info.RepositoryNames);
+                repositoryLookup.Add(info);
+            }
+
+            foreach (var info in repositoryLookup)
+            {
+                foreach (var item in info.DependencyNames)
+                {
+                    var dependency = repositoryLookup.Find(r => r.RepositoryNames.Contains(item));
+                    if (dependency != null)
+                    {
+                        // Testing has odd cyclic dependencies
+                        if (string.Equals(info.Name, "Testing", StringComparison.OrdinalIgnoreCase) &&
+                            string.Equals(dependency.Name, "aspnet.xunit", StringComparison.OrdinalIgnoreCase))
+                        {
+                            continue;
+                        }
+
+                        if (string.Equals(dependency.Name, "Testing", StringComparison.OrdinalIgnoreCase) &&
+                            (string.Equals(info.Name, "Common", StringComparison.OrdinalIgnoreCase) ||
+                            string.Equals(info.Name, "PlatformAbstractions", StringComparison.OrdinalIgnoreCase) ||
+                            string.Equals(info.Name, "Logging", StringComparison.OrdinalIgnoreCase)))
+                        {
+                            continue;
+                        }
+
+                        info.Dependencies.Add(dependency);
+                    }
+                }
+            }
+
+            return repositoryLookup.GroupBy(r => r.Order, r => r.Name).OrderBy(r => r.Key);
```

Comment:
ToList()/ToArray(), otherwise you'll reevaluate the output every time and I noticed that you do it at least twice (once to print the groups and once to build)